### PR TITLE
Onnxlstm

### PR DESCRIPTION
**Pull request by kali**

### CODE REVIEW

File: onnx/tests/onnx/mod.rs [hunk 1]
```diff
@@ -121,6 +119,8 @@ pub fn run_one<P: AsRef<path::Path>>(root: P, test: &str, optim: bool) {
                 );
             }
             for (ix, (a, b)) in computed.iter().zip(expected.iter()).enumerate() {
+//                println!("computed: {:?}", computed[ix].dump(true));
```
Comment (aldanor) [inline onnx/tests/onnx/mod.rs]: Temporary comments?

File: core/src/analyser/rules/proxies.rs [hunk 2]
```diff
@@ -129,18 +80,18 @@ impl Index<usize> for SharedTensorsProxy {
 /// solver.equals(input.rank, 2)
 /// solver.equals(input.shape[1], output.value[0][1])
 /// ```
-pub struct SharedTensorProxy {
+pub struct TensorProxy {
```
Comment (aldanor) [inline core/src/analyser/rules/proxies.rs]: So, `SharedTensorProxy` is now being replaced with tvec/slice being used directly?

File: onnx/tests/onnx/mod.rs (hunk 1 above)
Comment (kali) [reply onnx/tests/onnx/mod.rs]: Well this is test code, they come in handy when something does not work. I know it looks a bit dirty, but I haven't find a much better way.

File: core/src/analyser/rules/proxies.rs (hunk 2 above)
Comment (kali) [reply core/src/analyser/rules/proxies.rs]: There are two things here. 
First a bad name, coming from an over-enthusiast regexp. SharedTensorProxy should be TensorProxy.
Second, the SharedTensorsProxy is gone (with an S at tensors). The rationale being that the input and output arity is always known before we reach the inference stage, so we can work with a regular slice there. It simplifies a bit processing operators with big lists of arguments (like the classifier or the lstm, so i though it was time to do this).

**Comment by kali** [comment]
And thanks for having a look :)

**Comment by kali** [comment]
@aldanor I've split the TensorProxy changes in a different branch to be merged ASAP as it should save you work implementing the rules.

**Comment by aldanor** [comment]
Great, thanks! I would definitely prefer to rebase my branch early and in batches if possible :) (also it should make this PR much more readable after you merge that branch and rebase this one)

**Comment by kali** [comment]
Fully agree. I worked alone for a long time on this project, so I've been a bit sloppy, please bear with me.

**Comment by aldanor** [comment]
No worries :) (I have one such “project in Rust with huge scope, single developer” kind of project on top of me as well, so it’s fully understandable...)

**Comment by kali** [comment]
This branch was merged on the deepspeech branch.

**Pull request closed without merging**